src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -197,6 +221,7 @@ static void uct_cuda_copy_md_close(uct_md_h uct_md) {
 static uct_md_ops_t md_ops = {
     .close                  = uct_cuda_copy_md_close,
     .query                  = uct_cuda_copy_md_query,
+    .query_v2               = uct_cuda_copy_md_query_v2,
```

Comment:
for all transports:
we could remove `query` from ops at all and implement on top of `uct_cuda_copy_md_query_v2`.
otherwise, having both is a duplication.

Reply:
This was already done in https://github.com/openucx/ucx/pull/7847 but it meant making changes in existing implementation of query for transports or copying over attributes between v1 and v2. The main problem is that v2 attributes are flattened out so we have to add a function to copy between v1 and v2. This is why I chose to separate the calls and use the appropriate version as needed. Maybe we can eventually remove query v1?

Reply:
yes, we can remove `query_v1` from ops and implement `uct_md_query` on top `query_v2` op by requesting all attributes  by just copying them field by field.
it should be done only once and `query_v1` could be removed from transports.

Reply:
@dmitrygx should I remove all query_v1 instances in UCT in this PR?
cc @yosefe

Reply:
@Akshay-Venkatesh I think, yes. This is the same way how the following v2 APIs were introduced:
- `uct_md_mem_dereg_v2` - https://github.com/openucx/ucx/pull/6819
- `uct_md_mem_reg_v2` - https://github.com/openucx/ucx/pull/8250
- `uct_md_mkey_pack_v2 ` - https://github.com/openucx/ucx/pull/8072